### ISSUE TITLE
Automated upload of files from selected folders out of local OC folder

### ISSUE THREAD

**Issue by tobiasKaminsky** [owncloud/android#744]
Hi,

just an idea:
My whatsapp folder is growing and growing and consuming lots of (rare) memory.
So the idea is to extend the instantupload to watch any folder that we specify.

With this all whatsapp images/videos will automatically be uploaded to owncloud. When I have no memory left I can go to owncloud and say "delete local files" on the whatsapp folder.
(And via root explorer I can delete all folders/images in /data/whatsapp).

Later it would be great if the files are not copied but remain on the original storage path.
Then it would be possible to delete the whatsapp images from within owncloud app, which would be a great benefit for all users that do not have root or do not know that files are duplicated (but this is another issue, that already exists: #346 ).

What do you think? Or do you have a better idea to solve this?

PR: #1650

**Comment by Husky17** [comment]
Hi guys, I just wanted to jump in and say that I think the usefulness of this feature is understated. I think many others, myself included, would use features like this to sync all sorts of app documents and files that would normally be a pain to transfer from the device ownCloud storage. For example, geotracking files for running/biking trips, photos retouched using editor apps, records of sleep or gas apps. It would be awesome to get this abilty with the ownCloud client soon since it seemed so close.

Been using ownCloud and the clients for about a month now and its great, thanks for your contributions! :)

**Comment by tobiasKaminsky** [comment]
I will upload a PR the next days.
Maybe we can convince @jancborchardt to rate this higher ;)

**Comment by wolfen351** [comment]
Just want to chip in here too, +1 for this feature, came here specifically to see if there is a way to use a different camera path, and discovered this gem. It would be great to have a folder i know will sync to the server, and from the server too

**Comment by Bugsbane** [comment]
Really, this is not about pictures or videos or even InstantUpload at all, so much as asking for the same functionality of the desktop client to sync folders.

**Comment by wolfen351** [comment]
Agree, but as a side effect it will solve all these photo not uploading issues too :) (or at least provide a workaround)

**Comment by tobiasKaminsky** [comment]
@Bugsbane This is not about syncing. 
It is an extended instant upload that should work for any desired folder in the same way as the current instant uploader is working: from mobile phone to owncloud for any file type.

A real syncing is something complete different.

**Comment by Bugsbane** [comment]
Well, one way syncing for ofiles from a folder on your phone to a folder on the server. How is this different to your proposal? Admittedly the desktop client is two way, but that's the only dfifference I see, unless I'm missing something.

**Comment by tobiasKaminsky** [comment]
Yes, two sync vs one way sync is the one (and really big) difference as two way sync is much more complicated than just pushing all new files to the server.

**Comment by Bugsbane** [comment]
> two way sync is much more complicated than just pushing all new files to the server

OK, sure. I can agree with that.

**Comment by CoolCold** [comment]
@tobiasKaminsky should we make some donation gathering to let this be implemented? there are a lot of things which should be synced phone -> server (primarily for backup from my POV)

**Comment by tobiasKaminsky** [comment]
@CoolCold that would be cool. At least it will motivate me a bit more to finish the corresponding PR and push it into beta.

**Comment by helmo** [comment]
@CoolCold Let's do https://www.bountysource.com/issues/6434818-extend-instantupload-to-other-folders

**Comment by CoolCold** [comment]
@helmo I'm in, my payment has bee set up.

**Comment by davivel** [comment]
Jumping in. Having a look to current state from the user perspective.

**Comment by davivel** [comment]
My point on this is similar to what @jancborchardt already said before: there are other issues that we should address before. Some of them:

- #6, instant upload based of FileObserver, **just** on the default camera folder
- #827, set instant uploads per account when multiple accounts exist;
- #1136, automated folder two-way sync; (could go in parallel, if we limit the extended instant uploads to one-way sync; though I would prefer this #1136 goes first)

I understand how interesting is this feature to back up certain contents of the mobile device in our OC server. I must admit that it has won a lot of importance in my head with the time, I didn't see it that way in the past, but there have been too many smart comments in this repo about the topic to ignore them.

My concern is that we still don't sync good enough with a single mapping between device and server, name it mobile:[externalStorageRoot]/owncloud/[user]/ <----> server:[user]/ . If we start to add extra mappings we'll just spread the pending issues to more use cases. 

An easy-to-see example: without solving #827 first, we would automatically upload to different accounts (the last active in the app, whichever it is) not just the pictures or videos we take, but also the media files we receive in WhatsApp. They could go to our account in the office instead of our personal server. And I know what pictures I take, but don't know what media will receive. Not fun.

As @Bugsbane said, desktop client allows mapping different local folders to different server folders, but it's not a "free bar" feature, there are limitations (1). And it's not free of its own set of problems, even though our mates in the desktop client know much better how to sync (2). 

There are also very important UX concerns that need to be considered and discussed. Many mobile users don't even know that file browsers exist in smartphones. Really. Ask your friends, your relatives, it happens a lot. This is a feature for advanced users, and we should take the time to think how it can be offered and/or taught in a digestible way to common people (3). Maybe allowing any mapping is too much, or bring too many complications.


(1) Example: if we map the first local folder to the root folder of the server when we set-up the account in the client, we cannot map other local folders to server folders in that account.

(2) Good moment to notice: desktop client and mobile clients part from opposite sides as sync-files Vs access-files solutions, and both are converging to the middle. Some day we'll meet.

(3) Off-topic: https://www.youtube.com/watch?v=ainyK6fXku0#t=7s

**Comment by davivel** [comment]
If you don't mind, @tobiasKaminsky, I will rephrase the title

**Comment by xeyownt** [comment]
Can we have this issue bumped up?

I understand from the comments above that there are dependencies, but these do not seem to make much progress, and I would happily go for:
* **Issue with multiple accounts -> don't use multiple accounts**, or clearly state the limitations. In my android app, I was not even aware you could have multiple accounts. I have the feeling that multiple accounts setup are rather an exception.
* **Issue with two-way sync -> don't do two-way sync**. As stated above, one-way sync would be already a major benefit for the backup use case. One-way sync from/to any folder would be already a major improvement tmo.

**Comment by rijndael86** [comment]
any news on this feature? it's been some time :)

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by the Claude Code review agent.

This feature request — automatically uploading files from **arbitrary, user-selected local folders** (not just the camera/DCIM folder) — has long since been implemented in the ownCloud Android app.

The current **Auto Upload** feature (Settings → *Picture Uploads* / *Video Uploads*) lets you pick a custom **source folder** on the device via a folder picker and map it to a chosen remote target folder on the server, doing exactly the one-way "push everything from this local folder to ownCloud" behaviour discussed here. In the current codebase this is the configurable source path (e.g. `PREF__CAMERA_PICTURE_UPLOADS_SOURCE`) with its folder-picker selector in the automatic-uploads settings.

Since the requested capability now ships in the app, we are closing this as resolved/implemented. If you find the current Auto Upload doesn't cover a specific case you need, please open a fresh issue with up-to-date details against the current release. Thank you all for the discussion and patience over the years\!
